lib/active_record/connection_adapters/mysql2_ghost_adapter.rb

```diff
@@ -111,6 +144,25 @@ def schema_migration_update?(sql)
         INSERT_SCHEMA_MIGRATION_PATTERN =~ sql ||
           DROP_SCHEMA_MIGRATION_PATTERN =~ sql
       end
+
+      def build_add_index_sql(table_name, column_names, index_name, # rubocop:disable Metrics/ParameterLists
```

Comment:
inspired by the ActiveRecord code used to build a `CREATE INDEX` query (we can't use that because `gh-ost` only triggers on `ALTER TABLE` queries